### PR TITLE
added test for GDCblocksize

### LINKED ISSUE #893: C test for GCDblocksize() in pioc_sc.c
Need a C test for this.

### PR DESCRIPTION
**Pull request by edhartnett**
Added test and some documentation for this function. Still not clear what is going on - any comments where ??? are would be most appreciated.

Fixes #893.

### CODE REVIEW

File: src/clib/pioc_sc.c [hunk 1]
```diff
@@ -177,52 +186,60 @@ PIO_Offset GCDblocksize(int arrlen, const PIO_Offset *arr_in)
...
     }
 
+    /* ??? */
     numblks = numtimes + 1;
     if (numtimes == 0)
         n = numblks;
     else
         n = numtimes;
 
+    /* ??? */
```
Comment (jedwards4b) [inline src/clib/pioc_sc.c]: If numblks==1 then the result is arrlen and you can return.

File: src/clib/pioc_sc.c [hunk 2]
```diff
@@ -177,52 +186,60 @@ PIO_Offset GCDblocksize(int arrlen, const PIO_Offset *arr_in)
...
     bsize = (PIO_Offset)arrlen;
     if (numblks > 1)
     {
         PIO_Offset blk_len[numblks];
         PIO_Offset gaps[numtimes];
-        if(numtimes > 0)
+        if (numtimes > 0)
```
Comment (jedwards4b) [inline src/clib/pioc_sc.c]: If numblks > 1 then numtimes must be > 0 and this if block isn't needed.

File: src/clib/pioc_sc.c [hunk 3]
```diff
@@ -177,52 +186,60 @@ PIO_Offset GCDblocksize(int arrlen, const PIO_Offset *arr_in)
...
         {
             blk_len[i] = loc_arr[i] - loc_arr[i - 1];
             blklensum += blk_len[i];
         }
         blk_len[numblks - 1] = arrlen - blklensum;
 
+        /* Get the GCD in blk_len array. */
         bsize = lgcd_array(numblks, blk_len);
+
+        /* ??? */
```
Comment (jedwards4b) [inline src/clib/pioc_sc.c]: I don't recall why i needed these next two blocks, I remember struggling to get this right in all cases and I'm afraid that the end result is that bsize is almost always 1.

**Comment by edhartnett** [comment]
OK, thanks for the comments! I will add them to the PR...

**Pull request merged**